### PR TITLE
Modify handling of INStartVideoCallIntent and INStartAudioCallIntent …

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/AppDelegate.m [hunk 1]
```diff
@@ -359,6 +406,14 @@ - (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull N
             return NO;
         }
 
+        if ([Environment getCurrent].phoneManager.hasOngoingCall) {
+            DDLogWarn(@"%@ ignoring INStartAudioCallIntent due to ongoing RedPhone call.", self.tag);
+            return NO;
+        }
+        if ([Environment getCurrent].callService.call != nil) {
+            DDLogWarn(@"%@ ignoring INStartAudioCallIntent due to ongoing WebRTC call.", self.tag);
```
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: `return NO`?

Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: Good catch!

File: Signal/src/AppDelegate.m [hunk 2]
```diff
@@ -359,6 +406,15 @@ - (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull N
...
+            return NO;
+        }
+
```
Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: `OutboundCallInitiator` is asynchronous - i.e. it does the "supports webrtc?" request before actually starting a call. We might want to modify it to make this check as well before starting a call. Hmm, I guess I'll do that.

Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: @michaelkirk I've pushed another commit that does this.

Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: You mean make the "is busy?" check before starting the call?

Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: we seem to have some sort of concurrency bug in our GH commenting. 😉

**Pull request merged**